### ISSUE TITLE
False positive for indexing-exception

### ISSUE THREAD

**Issue by Mariatta** [PyCQA/pylint#3775]
### Steps to reproduce

I was getting false positive with pylint `indexing-exception [W1624] Indexing exceptions will not work on Python 3`
When I looked at the code, it is not a code for handling exception at all. It looks like the following:

The code looks sort of like this: (redacted)
```
    def find_entry_by_slug(self, entry):
        entries = self.client.entries(
            .....
        )
        if len(entries) > 0:
            article = entries[0].fields()['article']  # pylint: disable=indexing-exception
            .....
            return ....
        else:
            return None
```

It was complaining on the line where I added the `# pylint: disable` message.

Hope this helps.

### Expected behavior

It should not complain about the indexing-exception because the code is not handling any exception

### pylint --version output

pylint v 2.4.4

Thanks.

**Comment by dbaty** [comment]
I tried to fill in the blanks (by replacing "....." by dummy code and fixing other pylint warnings) but could not reproduce the issue with the latest version of pylint. Here is the code I tried with:

```python
# pylint: disable=missing-module-docstring,missing-function-docstring

def find_entry_by_slug(client):
    entries = client.entries()
    if len(entries) > 0:
        article = entries[0].fields()['article']
        return article
    return None
```

Would you be able to try and provide self-sufficient code with which the issue can be reproduced? Also, it would be useful to test with the latest version of pylint (2.6.0), if possible.

**Comment by PCManticore** [comment]
My impression is that `entries` is inferred as an exception, maybe something that is returned by the `client.entries()` call? Regardless without knowing more details, like the libraries involved and having a small reproduction example, there's not much we can do here to fix this. We can reopen once that is the case.
